packages/shared/createInjectionState/index.ts

```diff
@@ -18,6 +19,20 @@ export type CreateInjectionStateReturn<Arguments extends Array<any>, Return> = R
   useInjectedState: () => Return | undefined,
 ]>
 
+export interface CreateInjectionStateObject<Arguments extends Array<any>, Return> extends Record<string | number, unknown> {
```

Comment:
do we need to extend `Record` here?

the shape is defined by us so shouldn't need to be extensible afaict